modules/widgets/google-translate.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Plugin Name: Google Translate Widget for WordPress.com
+ * Plugin URI: http://automattic.com
+ * Description: Add a widget for automatic translation
+ * Author: Artur Piszek
+ * Version: 0.1
+ * Author URI: http://automattic.com
+ * Text Domain: jetpack
+ */
+
+class Google_Translate_Widget extends WP_Widget {
+	static $instance = null;
+
+	/**
+	 * Register widget with WordPress.
+	 */
+	function __construct() {
+		parent::__construct(
+			'google_translate_widget',
+			/** This filter is documented in modules/widgets/facebook-likebox.php */
+			apply_filters( 'jetpack_widget_name', __( 'Google Translate', 'jetpack' ) ),
+			array( 'description' => __( 'Automatic translation of your site content', 'jetpack' ) )
+		);
+		wp_register_script( 'google-translate-init', plugins_url( 'google-translate/google-translate.js', __FILE__ ) );
+		wp_register_script( 'google-translate', '//translate.google.com/translate_a/element.js?cb=googleTranslateElementInit', [ 'google-translate-init' ] );
+	}
+
+	/**
+	 * Display the Widget.
+	 *
+	 * @see WP_Widget::widget()
+	 *
+	 * @param array $args     Display arguments.
+	 * @param array $instance The settings for the particular instance of the widget.
+	 */
+	public function widget( $args, $instance ) {
+		// We never should show more than 1 instance of this.
+		if ( null === self::$instance ) {
+			/** This filter is documented in core/src/wp-includes/default-widgets.php */
+			$title = apply_filters( 'widget_title', $instance['title'] );
+			echo $args['before_widget'];
+			if ( ! empty( $title ) ) {
+				echo $args['before_title'] . esc_html( $title ) . $args['after_title'];
+			}
+			wp_localize_script( 'google-translate-init', '_wp_google_translate_widget', array( 'lang' => get_locale() ) );
+			wp_enqueue_script( 'google-translate-init' );
+			wp_enqueue_script( 'google-translate' );
+			echo '<div id="google_translate_element"></div>';
+			echo $args['after_widget'];
+			self::$instance = $instance;
+			// Admin bar is also displayed on top of the site which causes google translate bar to hide beneath.
+			// This is a hack to show google translate bar a bit lower.
+			if ( is_admin_bar_showing() ) {
+				echo '<style>.goog-te-banner-frame { top:32px !important } </style>';
```

Comment:
This could be added using `wp_add_inline_style` checking if `admin-bar` style was enqueued. A user might have dequeued that style and still have the setting to show admin bar enabled.
Also, using `wp_add_inline_style` allows to move this to a `wp_enqueue_scripts` hook.